overlord/snapstate/backend/export_test.go

```diff
@@ -73,3 +75,27 @@ func MockMkdirAllChown(f func(string, os.FileMode, sys.UserID, sys.GroupID) erro
 		mkdirAllChown = old
 	}
 }
+
+func MockSnapDataDirs(snap *snap.Info, opts *dirs.SnapDirOptions) ([]string, error) {
+	var found []string
+	for _, entry := range snap.DataHomeDirs(opts) {
+		entryPaths, err := filepath.Glob(entry)
+		if err != nil {
+			return nil, err
+		}
+		found = append(found, entryPaths...)
+	}
+	return found, nil
+}
+
+func MockSnapCommonDataDirs(snap *snap.Info, opts *dirs.SnapDirOptions) ([]string, error) {
+	var found []string
+	for _, entry := range snap.CommonDataHomeDirs(opts) {
+		entryPaths, err := filepath.Glob(entry)
+		if err != nil {
+			return nil, err
+		}
+		found = append(found, entryPaths...)
+	}
+	return found, nil
+}
```

Comment:
I'm not entirely understanding the reason for these functions or why they are defined in export_test.go as they are not actually needing things from the package

Reply:
Sorry for the confusion, I was under the impression based on this comment:


> > The main reason I am hesitant in adding unit tests for snapDataDirs/snapCommonDataDirs is because I will have to make both of them globally available on imports although their only use is within the backend package.
> 
> we usually use export_test.go to deal with this kind of question

That export_test.go would be the place for such implementations. Is there a specific way you would like me to implement this?

Reply:
@ghadi-rahme First of all, Thank you for you efforts!

Mocking functions is usually needed when trying to exercise a certain code path in a method that depend on other methods. This is done by injecting our own implementation in place of existing dependencies.
For example:
https://github.com/snapcore/snapd/blob/67e00993c2443f397d9f42b65eef24f41bea528d/overlord/snapstate/backend/export_test.go#L69-L75
https://github.com/snapcore/snapd/blob/67e00993c2443f397d9f42b65eef24f41bea528d/overlord/snapstate/backend/copydata_test.go#L1170-L1173

I believe what @pedronis meant was exporting them inside `export_test.go`, for example:
https://github.com/snapcore/snapd/blob/67e00993c2443f397d9f42b65eef24f41bea528d/overlord/snapstate/backend/export_test.go#L31-L35
In our case here that would be:
```
 var ( 
 	SnapCommonDataDirs = snapCommonDataDirs 
 	SnapDataDirs = snapDataDirs 
 ) 
```
Then you will be able to access non-exported functions in your tests.

Please let me know if I can help.

Reply:
thank you very much for the explanation, I will get to work on the implementation